From cac6062ad71718dff5e399093814cdd610f7e8c2 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 3 May 2018 11:07:58 +0300
Subject: [PATCH 0621/1051] arm64: dts: enable cp0 emmc on ARMADA 8KP boards

Enable 4-bit eMMC interface on CP0 for ARMADA 8KP boards
A8082,  A8084, A8162, A8164

Change-Id: Ic46126ab17eca284076ecb0dadada2b6c1869d99
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8082-db.dts | 7 +++++++
 arch/arm64/boot/dts/marvell/armada-8084-db.dts | 7 +++++++
 arch/arm64/boot/dts/marvell/armada-8162-db.dts | 7 +++++++
 arch/arm64/boot/dts/marvell/armada-8164-db.dts | 7 +++++++
 4 files changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db.dts b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
index 2a1a1f828544..92d1687b48a2 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
@@ -69,6 +69,13 @@
 	status = "disabled";
 };
 
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8084-db.dts b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
index fbb3647d51c0..4db1e55a7be7 100644
--- a/arch/arm64/boot/dts/marvell/armada-8084-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
@@ -77,6 +77,13 @@
 	status = "disabled";
 };
 
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8162-db.dts b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
index 51562e91e434..3728f03ac0a9 100644
--- a/arch/arm64/boot/dts/marvell/armada-8162-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
@@ -69,6 +69,13 @@
 	status = "disabled";
 };
 
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8164-db.dts b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
index bf72445821b3..ee97bd8533db 100644
--- a/arch/arm64/boot/dts/marvell/armada-8164-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
@@ -77,6 +77,13 @@
 	status = "disabled";
 };
 
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
-- 
2.17.1

